joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/datastructures/CSharpScope.scala

```diff
@@ -29,7 +30,11 @@ class CSharpScope(summary: CSharpProgramSummary)
   }
 
   override def isOverloadedBy(method: CSharpMethod, argTypes: List[String]): Boolean = {
-    method.parameterTypes.filterNot(_._1 == "this").size == argTypes.size
+    method.parameterTypes.filterNot(_._1 == "this").size == argTypes.size && method.parameterTypes
+      .filterNot(_._1 == "this")
+      .map(_._2)
+      .zip(argTypes)
+      .count({ case (x, y) => x != y }) == 0
```

Comment:
We can't make this guarantee because of polymorphism. I suggest we comment this out until the scope object can understand whether an argument is a valid subtype of a parameter

Reply:
If we comment this out, other methods (ex - Console.WriteLine) which have multiple overloads will pick up the first available type it gets from the TypeMap.

Reply:
Then I suggest we have some kind of fall-back, otherwise we're going to lose coverage. e.g.
1. Try match on exact types
2. If no result, then try match only on param count